ui/pages/confirmations/components/confirm/info/typed-sign/typed-sign.tsx

```diff
@@ -39,6 +43,14 @@ const TypedSignInfo: React.FC = () => {
         padding={2}
         marginBottom={4}
       >
+        {primaryType === EIP712_PRIMARY_TYPE_PERMIT && (
```

Comment:
I avoid writing local functions to return JSX, if lines of code increase a lot then I prefer creating child components.

Reply:
What I'm suggesting is to create child components, but they would exist in the same file as a means to break up and encapsulate the logic.

No different to using private methods or functions to break up a large code block, ultimately just private components.

Reply:
This file is small only 79 lines of code also I do not expect it to grow a lot. I prefer single small component per file.

Reply:
Understood, I appreciate there is an aspect of personal preference.

But purely as a discussion point, it would still technically be one component per file as the others wouldn't be exported, no different to exporting a single class supported by many local functions or private methods.

The goal is simply to add further readability to the JSX and spread out any rendering logic like the `isValidAddress` and `primaryType === `.

So it could look like:

```
<Box ...>
    <ApprovingToRow ... />
   <RequestFromRow ... />
   <InteractingWithRow ... />
   <Message ... />
</Box>
```

Reply:
May be for readability, but this is not a pattern I see very often in frontend code. It can also result in bit more props passing and increase in total lines of code.